### ISSUE TITLE
Checklist/Collections for Ornaments

### ISSUE THREAD

**Issue by joshhunt** [Bungie-net/api#840]
One thing that's quite difficult to show on DestinySets.com is what ornaments (and other resuablePlugItems) users have unlocked. My solution at the moment relies on the user having the 'host' item somewhere in their inventory, meaning many ornaments, especially on exotics, go missing on the site. Other reusablePlugItems, like Ghost Projections, suffer from this as well, but it's mitigated by the fact that users will always have a ghost in their inventory.

It would be very helpful if the API provided a way to tell me what ornaments (and other resuablePlugItems) users have unlocked, through a checklist or psuedo-collection, or similar.

**Comment by vthornheart-bng** [comment]
I dig that!  I'll have to look into it, but on the surface that sounds like a reasonable thing that we could pull out!

**Comment by afpac** [comment]
Might be too late to chime in on this, but it would be rad to include Emblem Variants in this as well. I can pull this information right now through a very unpleasant series of queries against the Collectibles endpoint, but making the ornaments checklist an "item variants" checklist and including Emblems in it as well might make sense.

**Comment by vthornheart-bng** [comment]
I dig that as well, if we can get the time to aggregate it together!  Ideally, if I can find a way to automate getting you that data, that'll be the most likely way we can make it happen.  I wouldn't be able to promise it if we end up having to upkeep manual configuration on our end... but my gut tells me that we'll be able to automate this, and if so then that makes it much more likely (when we get the time to do so!)

**Comment by joshhunt** [comment]
Whatever can be done would be greatly appreciated.

I'm hoping there's like a single source for unlockable reusablePlugItems somewhere that could be exposed as a checklist or something? 🤞

**Comment by vthornheart-bng** [comment]
Unfortunately not directly, but there may be some tricks I can do to infer it indirectly from the content.  That's my hope at least!

**Comment by floatingatoll** [comment]
This would be another place where filtering unacquired ornaments for only those that could be acquired _may_ be a relevant concern.

**Comment by vthornheart-bng** [comment]
True, and that is both a good point and probably something that would be extremely difficult for me to automate.  We have the ability to do that for Collectibles only because the designers explicitly added tests to each relevant collectible to determine visibility.  These ornaments definitely won't have that information.

I'll need to give this some thought.  The concerns about unacquired collectibles was a valid point, and this is too... and in this case I don't think I have an automate-able way to get that information (if I can even automate getting the list of them at all to begin with)

**Comment by ArkahnX** [comment]
Just chiming in, the game keeps track of what ornaments you own regardless of whether the item is in your inventory or in collections.

Shouldn't that mean there is some sort of database that holds ornament ownership status?

**Comment by vthornheart-bng** [comment]
(a good, specific example of how these flags and values are often unpredictable is actually something you can see in collections itself: if you start a new account, pay attention to when all of the green and blue items unlock.  You'll see that most of them unlock at once, in response to a single story mission being completed or level being reached, regardless of whether or not you actually acquired them!  It's a clever re-use of existing state by the folks who designed the Collections, to try and make a best guess as to when people would have been likely to see all these items: and in doing so, they saved a ton of storage for items that people more-or-less don't care about: but it also points out just how freely designers can associate state with behaviors in Destiny's content, even in a more "strictly defined" system like Collections!  You can imagine from there how more open-ended systems might behave.)

**Comment by ArkahnX** [comment]
I definitely remember this discussion before!

I can't say I wasn't hoping for a single easy to read spot for all of that data, but it definitely doesnt seem to be the case.

But, it did bring about another awesome "vthorn story" so I'd say that's a win

**Comment by vthornheart-bng** [comment]
Aye - I wish it was as well!  Though I am grateful that as much of this kind of evaluation exists in content as it does (as opposed to scripts).  I sometimes think about that - how much harder it would have been to have an API if *all* of this stuff happened/was defined in scripts instead of it being defined in content.  It's interesting how that initial engineering/design decision shaped the way the game was created - and in many ways, made the API something that could feasibly be worked on by just a small number of people (often one... though I really wish sometimes that there was more than one, or at least one doing it full time ;) I digress!).

**Comment by ckhicks** [comment]
We're glad you're the one...even if we all wish there were more, for your sake. XD
